### PR TITLE
Workflow for examples

### PR DESCRIPTION
**Pull request by amityadav0**
## Proposed Changes

+ Add workflow for examples
+ Add deploy scripts
+  Add run scripts
Note: some run scripts are not added because they depend on cases on how user interacts with the example, essential scripts are included for testing.

### CODE REVIEW

**Review by ratik21 — COMMENTED** [review]
each linter warning is being shown 2 times:
image

File: .github/workflows/examples.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.8"
+      - name: Install python dependencies
+        run: python -m pip install pipenv
+      - run: pipenv sync
+      - name: Install dependencies
+        run: pipenv run yarn install
+      - name: Execute lint
+        run: yarn lint
```
Comment (ratik21) [inline .github/workflows/examples.yaml:42]: Could it be possible we're seeing multiple lint warnings as we're executing lint in multiple workflows?

File: .github/workflows/examples.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+      - name: Setup Network
+        run: sudo apt-get update && sudo apt-get install -y gnupg2 curl software-properties-common
+      - run: curl -O https://releases.algorand.com/key.pub && sudo apt-key add key.pub
+      - run: sudo add-apt-repository "deb [arch=amd64] https://releases.algorand.com/deb/ stable main"
+      - run: sudo apt-get update && sudo apt-get install -y algorand
+      - run: cd infrastructure && make setup-private-net
+
+      - name: Example ASA
+        run: yarn link @algo-builder/algob  && cd ~/.yarn/bin && chmod +x algob
```
Comment (ratik21) [inline .github/workflows/examples.yaml:54]:
```suggestion
        run: yarn link @algo-builder/algob  && chmod +x ~/.yarn/bin/algob
```

Comment (ratik21) [reply .github/workflows/examples.yaml:54]: let's move this out of  `Example ASA` & move to `Setup Algob`

File: .github/workflows/examples.yaml (hunk 1 above)
Comment (amityadav0) [reply .github/workflows/examples.yaml:42]: maybe, do we need a lint in examples ? i think we can remove this

File: .github/workflows/examples.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      - name: Setup algob
+        run: yarn link @algo-builder/algob  && chmod +x ~/.yarn/bin/algob
+
+      - name: Example ASA 
+        run: cd examples/asa && pipenv run yarn algob deploy
+      - run: cd examples/asa && pipenv run yarn algob run scripts/transfer/gold-contract-sc.js
+      - run: cd examples/asa && pipenv run yarn algob run scripts/transfer/gold-delegated-lsig.js
+      - run: cd examples/asa && pipenv run yarn algob run scripts/transfer/gold-to-john.js
+      - run: cd examples/asa && pipenv run yarn algob run scripts/transfer/master-fund-john.js
+      - run: cd examples/asa && pipenv run yarn algob run scripts/transfer/tesla-to-john.js
```
Comment (ratik21) [inline .github/workflows/examples.yaml:60]: do we need to `cd` on each run? i was reading this [https://stackoverflow.com/a/58142276](https://stackoverflow.com/a/58142276). Can we use the same concept here?

Comment (amityadav0) [reply .github/workflows/examples.yaml:60]: yes we need it, ok will try this

Comment (amityadav0) [reply .github/workflows/examples.yaml]: no it doen;t work, it works only for a line under name section, so you have create more names. i think current one is good.

Comment (ratik21) [reply .github/workflows/examples.yaml:60]: ok

**Review by ratik21 — APPROVED** [review]

File: .github/workflows/examples.yaml (hunk 3 above)
Comment (amityadav0) [reply .github/workflows/examples.yaml]: trying run with `|` , this may work

Comment (amityadav0) [reply .github/workflows/examples.yaml]: works, thanks

**Pull request merged**